apps/passport-server/src/services/generic-issuance/GenericIssuanceService.ts

```diff
@@ -345,4 +348,43 @@ export class GenericIssuanceService {
   public async getEdgeCityBalances(): Promise<EdgeCityBalance[]> {
     return getEdgeCityBalances(this.context.dbPool);
   }
+
+  /**
+   * Given an email and order code, and optionally a pipeline ID (which defaults to DEVCON_PIPELINE_ID),
+   * returns the ticket previews for the given email and order code. A ticket preview is basically a
+   * PODTicket in non-pcd form - just the raw IPODTicketData. This is used to display all the tickets
+   * a user might need when trying to check into an event, without having them go through a costly
+   * and slow account registration flow.
+   */
+  public async handleGetTicketPreview(
+    email: string,
+    orderCode: string,
+    pipelineId?: string
+  ): Promise<TicketPreviewResultValue> {
+    const requestedPipelineId = pipelineId ?? process.env.DEVCON_PIPELINE_ID;
+    const pipeline = (await this.getAllPipelineInstances()).find(
+      (p) => p.id === requestedPipelineId && PretixPipeline.is(p)
+    ) as PretixPipeline | undefined;
+
+    if (!pipeline) {
+      throw new PCDHTTPError(
+        400,
+        "handleGetTicketPreview: pipeline not found " + requestedPipelineId
+      );
+    }
```

Comment:
will this break any existing 1-click links, i.e. do we handle the 400 on the client?

Reply:
i've now split the new one-click page out from the old one so that none of the new code interferes with the old flows